### PR TITLE
docs: Improvements to the bulk_update and bulk_destroy functions

### PR DESCRIPTION
**Pull request by simpers**
Improvements to the bulk_update and bulk_destroy functions of the Ash.Api module.

# Contributor checklist

- [ ] Bug fixes include regression tests
- [ ] Chores
- [x] Documentation changes
- [ ] Features include unit/acceptance tests
- [ ] Refactoring
- [ ] Update dependencies

### CODE REVIEW

File: lib/ash/api/api.ex [hunk 1]
```diff
@@ -1958,14 +1958,16 @@ defmodule Ash.Api do
...
               input :: map,
-              Keyword.t()
+              opts :: Keyword.t()
             ) ::
               Ash.BulkResult.t() | no_return
 
   @doc """
-  Updates all items in the provided enumerable or query with the provided input.
+  Updates all items in the provided enumerable or query with the provided input for the action, which
+  is a map, where the keys are the primary keys of the records to update, and the values are maps
```
Comment (zachdaniel) [inline lib/ash/api/api.ex:1969]: The keys to the input map aren't the primary keys of the records to update. That comes from the enumerable or query in the first argument.

File: lib/ash/api/api.ex [hunk 2]
```diff
@@ -1989,14 +1991,16 @@ defmodule Ash.Api do
...
               input :: map,
-              Keyword.t()
+              opts :: Keyword.t()
             ) ::
               Ash.BulkResult.t() | no_return
 
   @doc """
-  Destroys all items in the provided enumerable or query with the provided input.
+  Destroys all items in the provided enumerable or query with the provided input for the action, which
+  is a map, where the keys are the primary keys of the records to update, and the values are maps
```
Comment (zachdaniel) [inline lib/ash/api/api.ex:2002]: same here.

File: lib/ash/api/api.ex (hunk 1 above)
Comment (simpers) [reply lib/ash/api/api.ex:1969]: Okay, great! What does it mean "that comes"? Like, what could it look like? Will fix when I understand it better.

Comment (zachdaniel) [reply lib/ash/api/api.ex:1969]: What I mean is that with `bulk_update`, you don't provide primary keys, you provide a query or stream to be updated. For example:

```elixir
Resource
|> Ash.Query.filter(active == true)
|> MyApi.bulk_update(:deactivate)
```

or

```elixir
list_of_records = [%Resource{}, %Resource{}, %Resource{}]

MyApi.bulk_update(list_of_records, :deactivate)
```

Primary keys don't come into the equation in bulk updates or destroys.

Comment (simpers) [reply lib/ash/api/api.ex:1969]: Okay, so should the action need input, the input map will be the same input for all records in the list/stream?

Comment (zachdaniel) [reply lib/ash/api/api.ex:1969]: Correct.

Comment (simpers) [reply lib/ash/api/api.ex:1969]: Gotcha! Fixing :)

**Review by zachdaniel — APPROVED** [review]

**Pull request merged**